aggregator/src/aggregator.rs

```diff
@@ -2173,6 +2189,62 @@ impl VdafOps {
+                                let original_report_metadatas: Vec<_> = tx
+                                    .get_report_aggregations_for_aggregation_job(
+                                        vdaf.as_ref(),
+                                        &Role::Helper,
+                                        task.id(),
+                                        aggregation_job.id(),
+                                    )
+                                    .await?
+                                    .iter()
+                                    .map(|ra| ra.report_metadata())
+                                    .collect();
+                                let mutating_request_report_metadatas: Vec<_> = report_share_data
+                                    .iter()
+                                    .map(|rsd| rsd.report_share.metadata().clone())
+                                    .collect();
+                                let event = AggregationJobInitForbiddenMutationEvent {
+                                    aggregation_job_id: *aggregation_job.id(),
+                                    original_request_hash: existing_aggregation_job
+                                        .last_request_hash(),
+                                    original_report_metadatas,
+                                    original_batch_id: format!(
+                                        "{:?}",
+                                        existing_aggregation_job.partial_batch_identifier()
+                                    ),
+                                    original_aggregation_parameter: existing_aggregation_job
+                                        .aggregation_parameter()
+                                        .get_encoded()
+                                        .map_err(|e| datastore::Error::User(e.into()))?,
+                                    mutating_request_hash: aggregation_job.last_request_hash(),
+                                    mutating_request_report_metadatas,
+                                    mutating_request_batch_id: format!(
+                                        "{:?}",
+                                        aggregation_job.partial_batch_identifier()
+                                    ),
+                                    mutating_request_aggregation_parameter: aggregation_job
+                                        .aggregation_parameter()
+                                        .get_encoded()
+                                        .map_err(|e| datastore::Error::User(e.into()))?,
+                                };
+                                let event_id = crate::diagnostic::write_event(
+                                    log_forbidden_mutations,
+                                    "agg-job-illegal-mutation",
+                                    Box::new(event),
+                                )
+                                .map_err(|e| datastore::Error::User(e.into()))?;
```

Comment:
I think I'm fine with this, but on the edge. If the target directory is unwritable, e.g. out of disk space, then we'll be trickling up 500 for what amounts to an error encountered while logging. But this flag and situation have a narrow enough use case that we'll be watching it closely anyway.

Reply:
Also note that I don't think we log errors in transactions like this. So if something went wrong here the only indication will be at DEBUG level and a 500 error in the metrics. Can consider adding an `error!()` in the `.map_err()`.

Reply:
I think you're right. What I did is change the call to `write_event` so that if it fails, we log the error it returns but still move on in this function, so the client will still get HTTP 409. But I only did this for that call, as I think the other fallible calls in this block failing would be so unusual and rare that an HTTP 500 would be fine.

Reply:
I agree with _not_ failing out if we fail to write diagnostic information, i.e. we should log but continue on a failure to write diagnostic information.